chrome/content/zotero/xpcom/utilities_internal.js

```diff
@@ -2488,7 +2488,53 @@ Zotero.Utilities.Internal = {
 		}
 
 		return textContent;
-	}
+	},
+
+	/**
+	 * Costruct item title for a given attachment.
+	 * @param itemID - itemID of the attachment to be opened in the reader.
+	 */
+	async constructItemTitle(itemID) {
```

Comment:
This definitely isn't descriptive enough, but also I don't think this function needs to be in `Utilities.Internal`, since it's not a general-purpose function. Can we put it in `Zotero.Reader` as `getTabTitle()`? If that doesn't work, we could move it to `Zotero.Item` next to `getDisplayTitle()`and plan to support notes or other item types later.

Reply:
Gotcha, that makes sense. I suppose my thinking was that there may be a tab without a reader loaded in it, so I thought it would have to go somewhere that's not tied to the reader. And then I noticed `Zotero.Utilities.Internal.renderItemTitle` (which, granted, is way more general and used in a number of places) and thought to place this title-related function next to it. 

But there is no reason to not keep in `Zotero.Reader`, if that is more appropriate. I moved it to `Zotero.Reader.getDisplayTitle` with a few of the semantics fixes from above. There was also a small conflict, so I rebased because of it.

Reply:
Actually, maybe `Zotero.Item::getTabTitle()` makes more sense? We'll almost certainly support notes in tabs at some point, and calling a single method is simpler than adding a separate `getTabTitle()` to whatever code handles note tabs and calling each handler's `getTabTitle()` conditionally from each place we need to call this.

Reply:
Ok, yeah, if we want to account for notes in tabs for the future, `Zotero.Reader` would not be appropriate, since there would be no actual reader in the tab. So I moved it to `Zotero.Item` next to other title-related functions. I think it feels quite natural to have `Zotero.Item::getTabTitle()` next to `Zotero.Item::getDisplayTitle()`, actually!